src/components/ha-base-time-input.ts

```diff
@@ -249,13 +253,26 @@ export class HaBaseTimeInput extends LitElement {
               <mwc-list-item value="AM">AM</mwc-list-item>
               <mwc-list-item value="PM">PM</mwc-list-item>
             </ha-select>`}
+        ${this.clearable && !this.required && !this.disabled
+          ? html`<ha-icon-button
+              label="clear"
+              @click=${this._clearValue}
+              .path=${mdiClose}
+            ></ha-icon-button>`
+          : nothing}
       </div>
       ${this.helper
         ? html`<ha-input-helper-text>${this.helper}</ha-input-helper-text>`
         : ""}
     `;
   }
 
+  private _clearValue(): void {
+    fireEvent(this, "value-changed", {
+      value: null,
```

Comment:
Why `null` and `undefined`?
```suggestion
      value: undefined,
```

Reply:
I added a comment to the place where this is "caught". I can add it here as well.

Reply:
I don't think the consumer should deal with this, the input value is just cleared, so undefined.

Reply:
Ideally there would be some kind of value-cleared event that can be used.

Having a value-changed event without a value can be confusing.
So I googled a bit and using [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null) seems like a reasonable option here.

Reply:
I've removed the `null` value